torch/distributed/pipeline/sync/pipe.py

```diff
@@ -248,8 +240,9 @@ def __init__(
         if deferred_batch_norm:
             module = DeferredBatchNorm.convert_deferred_batch_norm(module, chunks)
 
-        self.partitions, self.devices = _split_module(module)
-        _verify_splitting(module, self.partitions, self.devices)
+        self.partitions = module
```

Comment:
could you elaborate how this change helps what you wanted to achieve in this PR? with _split_module, we had the flexibility of changing partitioning algorithm by modifying _split_module implementation.

Reply:
`split_module` was mostly redundant since we don't really split the module and in this API the module is already split by the user. Since this is an internal implementation detail, we can always change it later if needed.